src/mongocxx/test/spec/initial_dns_seedlist_discovery.cpp

```diff
@@ -0,0 +1,315 @@
+        }
+    });
+
+    mongocxx::options::tls tls_options;
+    mongocxx::options::client client_options;
+
+    auto ca_file_path = std::getenv("MONGOCXX_TEST_TLS_CA_FILE");
+    REQUIRE(ca_file_path);
+    tls_options.ca_file(ca_file_path);
+    tls_options.allow_invalid_certificates(true);
+    client_options.tls_opts(tls_options);
+    client_options.apm_opts(apm_opts);
+
+    try {
+        mongocxx::uri my_uri{test.uri};
+        class client client {
+            my_uri, test_util::add_test_server_api(client_options)
+        };
+        bool using_tls = client.uri().tls();
+        if (!using_tls) {
+            return;
+        }
+        REQUIRE(!test.error);
+        if (should_ping) {
+            validate_srv_max_hosts(client, test, mtx, new_hosts);
+            mongocxx::pool pool{my_uri,
+                                options::pool(test_util::add_test_server_api(client_options))};
+            auto pool_client = pool.acquire();
+            validate_srv_max_hosts(*pool_client, test, mtx, new_hosts);
+        }
+    } catch (mongocxx::exception& e) {
+        REQUIRE(test.error);
+    }
+}
+
+static void iterate_srv_max_hosts_tests(std::string dir, std::vector<std::string> files) {
+    for (const auto& file : files) {
+        auto test_doc = _doc_from_file("/" + dir + "/" + file);
+        auto test = InitialDNSSeedlistTest::parse(test_doc);
+        SECTION(file) {
+            run_srv_max_hosts_test_file(test);
+        }
+    }
+}
+
+static void assert_tls_enabled(void) {
+    using namespace mongocxx;
+    using bsoncxx::builder::basic::kvp;
+    using bsoncxx::builder::basic::make_document;
+
+    options::tls tls_options;
+    options::client client_options;
+
+    auto ca_file_path = std::getenv("MONGOCXX_TEST_TLS_CA_FILE");
+    REQUIRE(ca_file_path);
+    tls_options.ca_file(ca_file_path);
+    tls_options.allow_invalid_certificates(true);
+    client_options.tls_opts(tls_options);
+
+    try {
+        class mongocxx::client client {
+            uri{"mongodb://localhost:27017/?tls=true"},
+                test_util::add_test_server_api(client_options)
+        };
+
+        auto admin = client["admin"];
+        auto result = admin.run_command(make_document(kvp("ping", 1)));
+        REQUIRE(!result.empty());
+        REQUIRE(result.begin()->get_double() == 1.0);
+    } catch (mongocxx::operation_exception& e) {
+        FAIL("Unable to ping server with TLS. Is TLS enabled on the server?");
+    }
+}
+
+// TODO(CXX-2087) extend this test to run all Initial DNS Seedlist Discovery tests.
+TEST_CASE("uri::test_srv_max_hosts", "[uri]") {
+    mongocxx::instance::current();
+
+    if (!std::getenv("MONGOCXX_TEST_TLS_CA_FILE")) {
+        std::cerr << "The environment variable 'MONGOCXX_TEST_TLS_CA_FILE' must be present to run "
+                     "the test 'uri::test_srv_max_hosts'"
+                  << std::endl;
+        return;
+    }
```

Comment:
There's just a slight conflict between this suggestion and a previous suggestion from @kevinAlbs. I took out defining `MONGOCXX_TEST_TLS_CA_FILE` from this PR by Kevin's suggestion, which was to check if this variable was set in order to run the tests, and to leave it undefined for tests where TLS is not enabled. I'm fine with doing it either way, and this suggestion makes sense to me, but I would like to hear what Kevin would prefer.

Reply:
> which was to check if this variable was set in order to run the tests, and to leave it undefined for tests where TLS is not enabled

I do not think my suggestion conflicts with @kevinAlbs in intent. However, the use of TLS CA certificates is not exclusive to Initial DNS Seedlist Discovery tests. Therefore I suggest using a dedicated variable `MONGOCXX_TEST_DNS` to indicate whether the DNS tests should be executed rather than `MONGOCXX_TEST_TLS_CA_FILE`, and leaving `MONGOCXX_TEST_DNS` undefined instead. A ticket similar to CDRIVER-4230 can then refer to `MONGOCXX_TEST_DNS` regarding enabling of tests rather than `MONGOCXX_TEST_TLS_CA_FILE`.